### PR TITLE
Remove ledgers from ES

### PR DESCRIPTION
**Pull request by emord**
##### SUMMARY
This removes all the code associated with ledgers in ES. Discovered how unnecessary this was during my pillow optimization work and this should be the final PR in that series. This stops writing ledgers to ES. Actually removing the index can be done at any time after this is merged with `./manage.py prune_elastic_indices`.

The pillow still exists to compute automatic daily consumption and section entry combinations for exports.

Will save some processing on ES and ~100 GB on ICDS ES cluster (to be seen if that actually matters much)

### CODE REVIEW

**Review by snopoke — APPROVED** [review]

**Comment by snopoke** [comment]
Build failing:
```
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/mnt/commcare-hq-ro/corehq/apps/es/__init__.py", line 1, in <module>
    from . import (
ImportError: cannot import name 'ledgers'
```

File: corehq/pillows/ledger.py [hunk 1]
```diff
@@ -42,8 +42,6 @@ def _prepare_ledger_for_es(ledger):
 
     _update_ledger_section_entry_combinations(ledger)
 
-    return ledger
```
Comment (esoergel) [inline corehq/pillows/ledger.py]: This threw me at first - didn't expect `_prepare_ledger_for_es` to do a stateful operation.  I see now that `_update_ledger_section_entry_combinations` does, though.

**Review by esoergel — APPROVED** [review]
Nice commit structure here too.

**Comment by emord** [comment]
there's also some constants like LEDGER_INDEX_INFO and such I need to look into removing (they may not be harmful though

File: corehq/pillows/ledger.py (hunk 1 above)
Comment (emord) [reply corehq/pillows/ledger.py]: Not sure if you noticed but `_update_ledger_section_entry_combinations` was not the only stateful operation here. `_get_daily_consumption_for_ledger` also updates the daily consumption for the ledger https://github.com/dimagi/commcare-hq/blob/f453bc326a5f9571679bceeff2f9455ce1efee64/corehq/pillows/ledger.py#L57

This was also a surprise to me and I only found out about when I was removing tests in my original PR. For example: https://github.com/dimagi/commcare-hq/blob/f453bc326a5f9571679bceeff2f9455ce1efee64/corehq/apps/commtrack/tests/test_xml.py#L239

**Comment by ctsims** [comment]
> there's also some constants like LEDGER_INDEX_INFO and such I need to look into removing (they may not be harmful though

@emord  Setting up a fresh install environment today, and this definitely results in a blocking issue as far as I can tell

When running the pillowtop preindexing (`./manage.py ptop_preindex`), the system still thinks that it needs to preindex the `ledger` pillows since they are still in the [list of expected indices](https://github.com/dimagi/commcare-hq/blob/e94ad580c63dea5026570be42876719e36b794f2/corehq/pillows/utils.py#L93) but there's no factory to do so.

Traceback below
```
Traceback (most recent call last):
  File "./manage.py", line 195, in <module>
    main()
  File "./manage.py", line 41, in main
    execute_from_command_line(sys.argv)
  File "/home/ctsims/.virtualenvs/commcare-hq/lib/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/home/ctsims/.virtualenvs/commcare-hq/lib/python3.6/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/ctsims/.virtualenvs/commcare-hq/lib/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/ctsims/.virtualenvs/commcare-hq/lib/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/home/ctsims/commcare-hq/corehq/apps/hqcase/management/commands/ptop_preindex.py", line 121, in handle
    job.get()
  File "src/gevent/greenlet.py", line 683, in gevent._greenlet.Greenlet.get
  File "src/gevent/greenlet.py", line 317, in gevent._greenlet.Greenlet._raise_exception
  File "/home/ctsims/.virtualenvs/commcare-hq/lib/python3.6/site-packages/gevent/_compat.py", line 47, in reraise
    raise value.with_traceback(tb)
  File "src/gevent/greenlet.py", line 766, in gevent._greenlet.Greenlet.run
  File "/home/ctsims/commcare-hq/corehq/apps/hqcase/management/commands/ptop_preindex.py", line 53, in do_reindex
    FACTORIES_BY_SLUG[reindex_command](**kwargs).build().reindex()
KeyError: 'ledger-v1'
```

**Comment by emord** [comment]
@ctsims sorry about that. I thought I had only left the ones required for the pillow checkpoint. https://github.com/dimagi/commcare-hq/pull/25806 should fix the error you saw.

**Comment by ctsims** [comment]
@emord Thanks for the quick turnaround. I worked around it early by just removing ledgers from the index generation at https://github.com/dimagi/commcare-hq/pull/25806/files#diff-ebf9f4abd1070fc1bd1c557906d9090aL93, so seems likely that this PR would have fixed the issue but can't quite confirm since I don't have the replication anymore.

**Pull request merged**